src/enhanced/job/enhancedJobHandlers.ts

```diff
@@ -0,0 +1,56 @@
+import { JobHandler } from '../../job/jobHandler';
+import { ManifestJobHandler } from '../../job/manifestJobHandler';
+import { ProductionJobHandler } from '../../job/productionJobHandler';
+import { RegressionJobHandler } from '../../job/regressionJobHandler';
+import { StagingJobHandler } from '../../job/stagingJobHandler';
+
+/**
+ * This method overrides the JobHandler's current setEnvironmentVariablesMethod so that it points to
+ * the custom buckets that are used in the enhanced app.
+ * @param this reference to current object
+ */
+async function setEnvironmentVariablesEnhanced(this: JobHandler) {
```

Comment:
minor nitpick. would rename `this` parameter to something like `jobHandler`

Reply:
So, this is really weird, but I think the param name needs to be `this` to access properties of the object that are private or protected for that class. As soon as I change the parameter name as something other than `this` it says it can't access the properties of `_config` for example since that is a protected attribute.